# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/PostTraversalTask.cpp
# Commit: 095b3edec3c8
# Full Hash: 095b3edec3c8ba9318c71d0acc91e10ce57818dc
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2019-04-29 21:53:38
# Regressor Bug: 1514869
# File Overlap Count: 1
# Description:
#   Bug 1514869 - patch 2 - Adapt platform-font-list code to work with either the existing in-process font list or cross-process shared font list. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D22938
# ==============================================================================

diff -r 21ef00977ab6 -r 095b3edec3c8 layout/style/PostTraversalTask.cpp
--- a/layout/style/PostTraversalTask.cpp	Sat Apr 27 15:37:29 2019 +0000
+++ b/layout/style/PostTraversalTask.cpp	Sat Apr 27 15:37:58 2019 +0000
@@ -8,7 +8,10 @@
 
 #include "mozilla/dom/FontFace.h"
 #include "mozilla/dom/FontFaceSet.h"
-#include "gfxUserFontSet.h"
+#include "gfxPlatformFontList.h"
+#include "gfxTextRun.h"
+#include "ServoStyleSet.h"
+#include "nsPresContext.h"
 
 namespace mozilla {
 
@@ -37,6 +40,19 @@
     case Type::LoadFontEntry:
       static_cast<gfxUserFontEntry*>(mTarget)->ContinueLoad();
       break;
+
+    case Type::InitializeFamily:
+      Unused << gfxPlatformFontList::PlatformFontList()->InitializeFamily(
+          static_cast<fontlist::Family*>(mTarget));
+      break;
+
+    case Type::FontInfoUpdate:
+      nsPresContext* pc =
+          static_cast<ServoStyleSet*>(mTarget)->GetPresContext();
+      if (pc) {
+        pc->ForceReflowForFontInfoUpdate();
+      }
+      break;
   }
 }
 